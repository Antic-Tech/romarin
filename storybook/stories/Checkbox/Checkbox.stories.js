import { storiesOf } from '@storybook/react-native';
import React from 'react';
import CenterView from '../CenterView';
import Checkbox from './Checkbox';

storiesOf('Checkbox', module).addDecorator((getStory) => <CenterView>{getStory()}</CenterView>).add('default', () => (
	<React.Fragment>
		<Checkbox padding="Small" />
		<Checkbox checked />
	</React.Fragment>
));
