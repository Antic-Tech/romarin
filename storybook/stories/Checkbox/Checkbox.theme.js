export default (core) => ({
	checkboxSize: 28,
	checkboxBorderRadius: core.coreBorderRadius,
	checkboxColorChecked: core.colorPrimary,
	checkboxColorUnchecked: core.colorLighterGray,
	checkboxCheckColor: core.colorWhite,
	checkboxPadding: 8
});
