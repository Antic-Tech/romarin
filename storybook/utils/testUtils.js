import { mount, shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

export const renderComponent = (jsx, { shallowRender = true, store } = {}) => {
	const renderOptions = { context: { store } };

	if (shallowRender) {
		return shallow(jsx, renderOptions);
	}

	return mount(jsx, renderOptions);
};

export const componentFactory = Component => (props = {}, options = {}) => {
	return renderComponent(<Component {...props} />, options);
};

export const getTree = jsx => renderer.create(jsx).toJSON();

export const verifySnapshot = jsx => {
	const tree = getTree(jsx);

	expect(tree).toMatchSnapshot();
};
