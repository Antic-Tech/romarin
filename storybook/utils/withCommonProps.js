import React from 'react';
import { StyleSheet } from 'react-native';
import ThemeProvider from '../theme';

const Sizes = {
	XXSmall: 'XXSmall',
	XSmall: 'XSmall',
	Small: 'Small',
	Medium: 'Medium',
	Large: 'Large',
	XLarge: 'XLarge',
	XXLarge: 'XXLarge'
};

const withPadding = (Component) =>
	class WithLoading extends React.Component {
		render() {
			const {
				style,
				padding,
				paddingHorizontal,
				paddingVertical,
				paddingTop,
				paddingBottom,
				paddingLeft,
				paddingRight,
				...props
			} = this.props;

			const newStyles = StyleSheet.flatten([
				{
					padding: padding ? ThemeProvider.theme[`corePadding${padding}`] : undefined,
					paddingHorizontal: paddingHorizontal
						? ThemeProvider.theme[`corePadding${paddingHorizontal}`]
						: undefined,
					paddingVertical: paddingVertical ? ThemeProvider.theme[`corePadding${paddingVertical}`] : undefined,
					paddingTop: paddingTop ? ThemeProvider.theme[`corePadding${paddingTop}`] : undefined,
					paddingBottom: paddingBottom ? ThemeProvider.theme[`corePadding${paddingBottom}`] : undefined,
					paddingLeft: paddingLeft ? ThemeProvider.theme[`corePadding${paddingLeft}`] : undefined,
					paddingRight: paddingRight ? ThemeProvider.theme[`corePadding${paddingRight}`] : undefined
				},
				style
			]);

			return <Component style={newStyles} {...props} />;
		}
	};

const withMargin = (Component) =>
	class WithLoading extends React.Component {
		render() {
			const {
				style,
				margin,
				marginHorizontal,
				marginVertical,
				marginTop,
				marginBottom,
				marginLeft,
				marginRight,
				...props
			} = this.props;

			const newStyles = StyleSheet.flatten([
				{
					margin: margin ? ThemeProvider.theme[`coreMargin${margin}`] : undefined,
					marginHorizontal: marginHorizontal
						? ThemeProvider.theme[`coreMargin${marginHorizontal}`]
						: undefined,
					marginVertical: marginVertical ? ThemeProvider.theme[`coreMargin${marginVertical}`] : undefined,
					marginTop: marginTop ? ThemeProvider.theme[`coreMargin${marginTop}`] : undefined,
					marginBottom: marginBottom ? ThemeProvider.theme[`coreMargin${marginBottom}`] : undefined,
					marginLeft: marginLeft ? ThemeProvider.theme[`coreMargin${marginLeft}`] : undefined,
					marginRight: marginRight ? ThemeProvider.theme[`coreMargin${marginRight}`] : undefined
				},
				style
			]);

			return <Component style={newStyles} {...props} />;
		}
	};

const withCommonProps = (Component) => withPadding(withMargin(Component));

export { withPadding, withMargin, withCommonProps };
