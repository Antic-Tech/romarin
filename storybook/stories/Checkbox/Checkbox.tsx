// import Icon from '@components/Icon/Icon.component';
import React from 'react';
import { Image, View } from 'react-native';
import ThemeProvider from '../../theme';
import { withCommonProps } from '../../utils/withCommonProps';

export default withCommonProps(
	class Checkbox extends React.Component {
		render() {
			const { checked } = this.props;

			return (
				<View
					style={{
						borderRadius: ThemeProvider.theme.checkboxBorderRadius,
						borderColor: checked
							? ThemeProvider.theme.focusedInputBorderColor
							: ThemeProvider.theme.inputBorderColor,
						height: ThemeProvider.theme.checkboxSize,
						width: ThemeProvider.theme.checkboxSize,
						backgroundColor: checked ? ThemeProvider.theme.checkboxColorChecked : undefined,
						borderWidth: ThemeProvider.theme.inputBorderWidth,
						justifyContent: 'center',
						alignItems: 'center'
					}}
				>
					{checked && (
						<Image
							style={{
								backgroundColor: 'transparent',
								height: ThemeProvider.theme.checkboxSize - ThemeProvider.theme.checkboxPadding,
								width: ThemeProvider.theme.checkboxSize - ThemeProvider.theme.checkboxPadding
							}}
							resizeMode="cover"
							source={require('./check.png')}
						/>
					)}
				</View>
			);
		}
	}
);
