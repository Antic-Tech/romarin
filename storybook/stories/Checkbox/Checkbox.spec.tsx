import React from 'react';
import { verifySnapshot } from '../../utils/testUtils';
import Checkbox from './Checkbox';

describe('Checkbox', () => {
	describe('Snapshots', () => {
		it('renders correctly', () => {
			verifySnapshot(<Checkbox />);
			verifySnapshot(<Checkbox checked />);
		});
	});
});
