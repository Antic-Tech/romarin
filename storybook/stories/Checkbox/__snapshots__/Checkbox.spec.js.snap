// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Snapshots renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderColor": "#dedede",
      "borderRadius": 10,
      "borderWidth": 1,
      "height": 28,
      "justifyContent": "center",
      "width": 28,
    }
  }
/>
`;

exports[`Checkbox Snapshots renders correctly 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2C7CF6",
      "borderColor": "#2C7CF6",
      "borderRadius": 10,
      "borderWidth": 1,
      "height": 28,
      "justifyContent": "center",
      "width": 28,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../storybook/stories/Checkbox/check.png",
      }
    }
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 20,
        "width": 20,
      }
    }
  />
</View>
`;
